docs/tutorials/gluon/info_gan.md

```diff
@@ -0,0 +1,436 @@
+generator = Generator()
+generator.hybridize()
+generator.initialize(mx.init.Normal(0.002), ctx=ctx)
+
+discriminator = Discriminator()
+discriminator.hybridize()
+discriminator.initialize(mx.init.Normal(0.002), ctx=ctx)
+
+lr   = 0.0001
+beta = 0.5
+
+g_trainer = gluon.Trainer(generator.collect_params(), 'adam', {'learning_rate': lr, 'beta1': beta})
+d_trainer = gluon.Trainer(discriminator.collect_params(), 'adam', {'learning_rate': lr, 'beta1': beta})
+q_trainer = gluon.Trainer(discriminator.Q.collect_params(), 'adam', {'learning_rate': lr, 'beta1': beta})
+```
+
+Create vectors with real (=1) and fake labels (=0).
+
+
+```python
+real_label = nd.ones((batch_size,), ctx=ctx)
+fake_label = nd.zeros((batch_size,),ctx=ctx)
+```
+
+Load a pretrained model.
+
+
+```python
+if os.path.isfile('infogan_d_latest.params') and os.path.isfile('infogan_g_latest.params'):
+    discriminator.load_parameters('infogan_d_latest.params', ctx=ctx, allow_missing=True, ignore_extra=True)
+    generator.load_parameters('infogan_g_latest.params', ctx=ctx, allow_missing=True, ignore_extra=True)
+```
+
+The latent code is part of the Generator input and it contains mutliple variables (continuous, categorical) that can represent different distributions. In order to make sure that the Generator uses the latent code, mutual information is introduced into the GAN loss term. Mutual information measures how much X is known given Y or vice versa. It is defined as:
+
+![gif](https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/doc/tutorials/info_gan/entropy.gif) 
+
+The InfoGAN loss is:
+
+![gif](https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/doc/tutorials/info_gan/loss.gif)
+
+where `V(D,G)` is the GAN loss and the mutual information `I(c, G(z, c))` goes in as regularization. The goal is to reach high mutual information, in order to learn meaningful codes for the data. 
+
+
+Define the loss functions. `SoftmaxCrossEntropyLoss` for the categorical code,  `L2Loss` for the continious code and `SigmoidBinaryCrossEntropyLoss` for the normal GAN loss.
+
+
+```python
+loss1 = gluon.loss.SigmoidBinaryCrossEntropyLoss()
+loss2 = gluon.loss.L2Loss()
+loss3 = gluon.loss.SoftmaxCrossEntropyLoss()
+```
+
+This function samples `c`, `z`, and concatenates them to create the generator input.
+
+
+```python
+def create_generator_input():
+    
+    #create random noise
+    z      = nd.random_normal(0, 1, shape=(batch_size, z_dim), ctx=ctx)
+    label  = nd.array(np.random.randint(n_categories, size=batch_size)).as_in_context(ctx)
+    c1     = nd.one_hot(label, depth=n_categories).as_in_context(ctx)
+    c2     = nd.random.uniform(-1, 1, shape=(batch_size, n_continuous)).as_in_context(ctx)
+
+    # concatenate random noise with c which will be the input of the generator
+    return nd.concat(z, c1, c2, dim=1), label, c2
+```
+
+Define the training loop. 
+1. The discriminator receives `real_data` and `loss1` measures how many real images have been identified as real
```

Comment:
The fact that Discriminator accepts `real data` along with the `fake data` generated by the Generator is currently not reflected in the architecture diagram. Can it be updated?

Reply:
Thanks for the review! I added your comments. I will also update the diagram as you suggested and upload it to dmlc/webdata in the the coming days.